tests/python/pants_test/backend/python/interpreter_selection_utils.py

```diff
@@ -107,3 +110,48 @@ def skip_unless_python27_and_python3_present(func):
 def skip_unless_python27_and_python36_present(func):
   """A test skip decorator that only runs a test method if python2.7 and python3.6 are present."""
   return skip_unless_all_pythons_present(PY_27, PY_36)(func)
+
+
+def skip_if_interpreter_is_any_python_version(*versions):
+  """A decorator that skips if the current interpreter version is any of the of the specified versions.
+
+  :param string *versions: Python version strings, such as "2.7", "3".
+  """
+  current_interpreter = ".".join(map(str, sys.version_info[0:2]))
+  # NB: if only the major Python version is specified, e.g. `2` or `3`, we must append `.*` to the string
+  # for packaging.specifiers.SpecifierSet to work properly.
+  version_strings = [
+    "=={}.*".format(version) if len(version) == 1 else "=={}".format(version)
+    for version in versions
+  ]
+  valid_versions = SpecifierSet(*version_strings)
+  return (
+    skipIf(True, "Current interpreter is one of the specified Python versions.")
+    if current_interpreter in valid_versions
+    else skipIf(False, "Current interpreter not one of the specified Python versions.")
+  )
+
+
+def skip_if_python2(func):
+  """A test skip decorator that skips if the current interpreter is Python 2."""
+  return skip_if_interpreter_is_any_python_version(PY_2)(func)
+
+
+def skip_if_python27(func):
+  """A test skip decorator that skips if the current interpreter is Python 2.7."""
+  return skip_if_interpreter_is_any_python_version(PY_27)(func)
+
+
```

Comment:
So, I think that the existing decorators in this file attempt to locate interpreters on the system: that is, they determine whether an interpreter "is present".

The new ones you've added skip if pytest is running inside of a particular interpreter (which you call "current" in the comments).

The two implementations have different usages: the former is for integration tests (usually?), while the latter are very much for unit tests (since a subprocessed pants will not necessarily use the same interpreter as the parent).

Because of this difference, it might be good to put the two sets of decorators in different places?

Reply:
> The two implementations have different usages: the former is for integration tests (usually?), while the latter are very much for unit tests (since a subprocessed pants will not necessarily use the same interpreter as the parent).

Not necessarily. The intended use is to not run the integration test to begin with. See https://github.com/pantsbuild/pants/pull/7261/files#diff-c624152df01281abd123b6e962f5f5f9R18 for the original motivation. We never get to the point of having to inspect which interpreter the subprocess will run with - we simply skip the test if using Python 3.7. This is useful for both unit and integration tests.

Thus, I would advocate keeping these in the same file, though I can still move it if you'd like? If so, which file do you recommend using? Create a new file in `testutils`? (Speaking of which, I think this file actually belongs in `testutils` and not `backend/python`, regardless of what we decide to do with this PR).

Reply:
Commented there... I think that just using the one-liner for now is fine.

Reply:
Regarding integration vs unit: I was referring more specifically to `PantsRunIntegrationTest` (which subprocesses pants) vs `TestBase` (which doesn't subprocess)... the case on #7261 is different, but shaped more like the latter case.